services/provider/server/server.go

```diff
@@ -694,21 +703,84 @@ func (s *OCSProviderServer) getOCSSubscriptionChannel(ctx context.Context) (stri
 		klog.Info("unable to find ocs-operator subscription")
 		return "", nil
 	}
-	if subscription.Status.InstalledCSV == "" {
+
+	return s.getLastKnownValidChannel(ctx, subscription)
```

Comment:
As mentioned offline, there is no real validity to split the functions here, this outer functions do nothing but read the subscription and the new function cannot assume a subscription as an argument because we only store a single "last known" set of values in memory which are tied to a very specific subscription. 

Either we generalize and start holding a set of values and a mutex for any given subscription (I think this is not necessary and an overkill) or merge the 2 functions back into one 

A third option is to keep the generalised function (which takes a subscription) but move all caching concerns, including the short circuiting of a read from the  cache and the "last known" value updates here

Reply:
Updated the logic in a different way Keeping these things in mind.

- Even if we remove the caching mechanism the code still works correctly
- The whole block of logic all it does is validate if the channel is valid. Named it so structured it so.
- In case any error was faced during the sub channel validation phase a empty channel along with a simple error is sent out.